robottelo/vm.py

```diff
@@ -498,6 +506,18 @@ def register_contenthost(self, org, activation_key=None, lce=None,
             )
             if auto_attach:
                 cmd += u' --auto-attach'
+        elif consumerid is not None:
```

Comment:
This conditional will also pass for ```"", [], {}``` and other "empty" values you may not want. 
To avoid this, the conditional could simply be changed to ```elif consumerid:```

Reply:
@JacobCallahan  also agree, but I did it this way because I want continue with same code style as in these [lines right above](https://github.com/SatelliteQE/robottelo/pull/7623/files#diff-2be7fd316a53f5055bef542dab4c3ad7R487-R505)